cmd/vic-machine/configure/configure.go

```diff
@@ -204,6 +204,7 @@ func (c *Configure) copyChangedConf(o *config.VirtualContainerHostConfigSpec, n
 	if c.OpsCredentials.IsSet {
 		o.Username = n.Username
 		o.Token = n.Token
+		o.GrantPermsLevel = n.GrantPermsLevel
```

Comment:
I'm not sure whether this is the behavior we want. (I'm also not sure that it isn't.)

By including `o.GrantPermsLevel = n.GrantPermsLevel` in this check, if an administrator uses `vic-machine configure` to change an operations user's password which previously had a `GrantPermsLevel` of `AddPerms` _without_ explicitly including `--ops-grant-perms` in that command we'll actually _downgrade_ `GrantPermsLevel` to `""`. I think that would be surprising behavior: change the password (e.g., due to expiration) and no longer have permissions be automatically managed.

It may be better to have a check here along the lines of `clic.IsSet("ops-grant-perms")` so that we only adjust `GrantPermsLevel` if `--ops-grant-perms` or `--ops-grant-perms=false` is included on the command line. (To do this, you'd just need to pass `clic *cli.Context` into the `copyChangedConf` method.) In that case, we only change the properties a user has asked us to change.

A downside of both the current implementation and this suggestion (and therefore an upside of the proposed change) is that if you change to a completely new operations user we won't clear the `GrantPermsLevel`. This could _also_ be surprising: this is a case where the user might _expect_ us to change something even though they haven't asked us to (because they may see `--ops-grant-perms` as being associated with the user, not the VCH). Unfortunately, the way `configure` is designed really doesn't give us a lot of options here (and leads to similar issues for other settings).

Reply:
Good catch. There was no clean way to get the name of the option out of the cli flags, nor did there seem to be much reason to declare a constant for it all on its own, but if you find it irksome, I can pull it out as a constant anyway.